src/ChainedMap.js

```diff
@@ -82,13 +118,43 @@ module.exports = class extends Chainable {
       }, {});
   }
 
-  when(condition, trueBrancher = Function.prototype, falseBrancher = Function.prototype) {
+  when(condition, whenTruthy = Function.prototype, whenFalsy = Function.prototype) {
     if (condition) {
-      trueBrancher(this);
+      whenTruthy(this);
     } else {
-      falseBrancher(this);
+      whenFalsy(this);
     }
 
     return this;
   }
+}
+
+ChainedMap.orderable = (Class) => class extends Class {
+  before(name) {
+    if (this.__after) {
+      throw new Error(`Unable to set before(${JSON.stringify(name)}) with existing value for after`);
+    }
+
+    this.__before = name;
+    return this;
+  }
+
+  after(name) {
+    if (this.__after) {
+      throw new Error(`Unable to set after(${JSON.stringify(name)}) with existing value for before`);
+    }
+
+    this.__after = name;
+    return this;
+  }
+
+  merge(obj, omit = []) {
+    if (obj.before && obj.after) {
+      throw new Error(`Unable to merge before: ${JSON.stringify(obj.before)} and after: ${JSON.stringify(obj.after)} on the same object`);
```

Comment:
Can this case be hit given the checks at time of `before()` and `after()` use?

Reply:
Yes, when merging in configuration from an object, these can be set by the user:

```js
config.merge({
  module: {
    rule: {
      compile: {
        use: {
          cache: {
            loader: 'cache-loader',
            before: 'babel'
          }
        }
      }
    }
  }
})
```